File: go/p007.go
package main

import "fmt"

func prime(a int) bool {
	if a < 2 {
		return false
	}
	for i := 2; i*i <= a; i++ {
		if a%i == 0 {
			return false
		}
	}
	return true
}

func main() {
	count := 0
	for i := 1; ; i++ {
		if prime(i) {
			count += 1
		}
		if count == 10001 {
			fmt.Println(i)
			// output: 104743
			break
		}
	}
}

File: rust/p007.rs
fn prime(a: int) -> bool {
    if a < 2 {
        return false;
    }
    let mut i = 2;
    while i*i <= a {
        if a%i == 0 {
            return false;
        }
        i += 1;
    }
    return true;
}

fn main() {
    let mut count = 0;
    let mut i = 1;
    loop {
        if prime(i) {
            count += 1;
        }
        if count == 10001 {
            println(fmt!("%?",i));
            // output: 104743
            break;
        }
        i += 1;
    }
}
